File: app/src/main/java/ru/samtakoy/core/services/learn_courses/LearnsApi.java
package ru.samtakoy.core.services.learn_courses;

import android.app.PendingIntent;

import org.jetbrains.annotations.NotNull;

import javax.inject.Inject;

import ru.samtakoy.R;
import ru.samtakoy.core.database.room.entities.types.LearnCourseMode;

import static ru.samtakoy.core.services.NotificationsConst.NOTIFICATION_ID_NEW_LEARNINGS_AVAILABLE;
import static ru.samtakoy.core.services.NotificationsConst.REQ_CODE_NEW_LEARNINGS_ALARM;
import static ru.samtakoy.core.services.NotificationsConst.REQ_CODE_NEW_LEARNINGS_CANCEL;
import static ru.samtakoy.core.services.NotificationsConst.REQ_CODE_NEW_LEARNINGS_CLICK;

public class LearnsApi extends CoursesApi {

    @Inject
    public LearnsApi() {
    }

    @Override
    @NotNull
    protected LearnCourseMode getLearnCourseMode() {
        return LearnCourseMode.LEARN_WAITING;
    }

    public void shiftLearnCourses() {
        shiftLearnCourses(getLearnCourseMode(), getShiftMillis());
        rescheduleNewLearnCourses();
    }

    /** обновить нотификации и таймеры напоминания */
    public void rescheduleNewLearnCourses(){

        LearnCourseMode lcMode = getLearnCourseMode();
        int notificationId =       NOTIFICATION_ID_NEW_LEARNINGS_AVAILABLE;
        int notificationIconId =   android.R.drawable.ic_menu_report_image;
        String notificationTitle = getString(R.string.notifications_learn_waitings_title);
        String notificationText =  getString(R.string.notifications_learn_waitings_text);

        PendingIntent clickIntent = PendingIntent.getService(mContext, REQ_CODE_NEW_LEARNINGS_CLICK, getShowUiIntent(), 0);
        PendingIntent cancelIntent = PendingIntent.getService(mContext, REQ_CODE_NEW_LEARNINGS_CANCEL, getShiftIntent(), 0);

        rescheduleLearnCourses(lcMode, notificationId, notificationIconId, notificationTitle, notificationText, clickIntent, cancelIntent);
    }

    @Override
    protected PendingIntent getLearnCoursesAlarmPendingIntent(boolean noCreate){
        return PendingIntent.getService(mContext, REQ_CODE_NEW_LEARNINGS_ALARM, getReSchedulingIntent(), noCreate ? PendingIntent.FLAG_NO_CREATE : 0);
    }


}

File: app/src/main/kotlin/ru/samtakoy/features/notifications/learn_courses/LearnsApi.kt
package ru.samtakoy.features.notifications.learn_courses

import android.app.PendingIntent
import android.content.Context
import ru.samtakoy.R
import ru.samtakoy.common.resources.Resources
import ru.samtakoy.data.learncourse.CoursesRepository
import ru.samtakoy.domain.learncourse.LearnCourseMode
import ru.samtakoy.features.notifications.NotificationsConst

// TODO тут все пересмотреть и переделать
class LearnsApi(
    context: Context,
    resources: Resources,
    coursesRepository: CoursesRepository,
) : CoursesApi(
    context,
    resources,
    coursesRepository,
) {
    override val learnCourseMode: LearnCourseMode
        get() = LearnCourseMode.LEARN_WAITING

    fun shiftLearnCourses() {
        shiftLearnCourses(learnCourseMode, shiftMillis)
        rescheduleNewLearnCourses()
    }

    /** обновить нотификации и таймеры напоминания  */
    fun rescheduleNewLearnCourses() {
        val lcMode = learnCourseMode
        val notificationId = NotificationsConst.NOTIFICATION_ID_NEW_LEARNINGS_AVAILABLE
        val notificationIconId = android.R.drawable.ic_menu_report_image
        val notificationTitle = resources.getString(R.string.notifications_learn_waitings_title)
        val notificationText = resources.getString(R.string.notifications_learn_waitings_text)

        val clickIntent =
            PendingIntent.getService(
                context,
                NotificationsConst.REQ_CODE_NEW_LEARNINGS_CLICK,
                showUiIntent,
                PendingIntent.FLAG_IMMUTABLE
            )
        val cancelIntent =
            PendingIntent.getService(
                context,
                NotificationsConst.REQ_CODE_NEW_LEARNINGS_CANCEL,
                shiftIntent,
                PendingIntent.FLAG_IMMUTABLE
            )

        rescheduleLearnCourses(
            lcMode,
            notificationId,
            notificationIconId,
            notificationTitle,
            notificationText,
            clickIntent,
            cancelIntent
        )
    }

    override fun getLearnCoursesAlarmPendingIntent(noCreate: Boolean): PendingIntent {
        return PendingIntent.getService(
            context,
            NotificationsConst.REQ_CODE_NEW_LEARNINGS_ALARM,
            reSchedulingIntent,
            if (noCreate) PendingIntent.FLAG_NO_CREATE else 0
        )
    }
}
